src/newsimplification/Simplifier.ts

```diff
@@ -0,0 +1,150 @@
+import { CommonModel, Schema } from '../models';
+import { SimplificationOptions } from '../models/SimplificationOptions';
+import { simplifyName, isModelObject } from './Utils';
+import simplifyProperties from './SimplifyProperties';
+import simplifyAllOf from './SimplifyAllOf';
+
+export class Simplifier {
+  static defaultOptions: SimplificationOptions = {
+    allowInheritance: false
+  }
+
+  private anonymCounter = 1;
+  private seenSchemas: Map<Schema, CommonModel> = new Map();
+  private iteratedModels: Record<string, CommonModel> = {};
+  
+  constructor(
+    readonly options: SimplificationOptions = Simplifier.defaultOptions,
+  ) {
+    this.options = { ...Simplifier.defaultOptions, ...options };
+  }
+
+  /**
+   * Simplifies a schema into instances of CommonModel. 
```

Comment:
I think that it is good idea to have here a detailed description what simplify is actually doing. I do not get this, it is too unclear for me.

Reply:
I agree that the `simplifier` function is a bit much... I was thinking it might be time to add a pre-processor where it iterates the simplified CommonModels and then split them appropriately. 

What if we restructure the simplifier to simply be an input processor for JSON Schema draft 7 inputs (since this is really what it is. This way adding a pre-processor also makes sense, and we can reduce the complexity behind the `simplify` function to only return a single `CommonModel` instead of multiple.

Thoughts @czlowiek488 ?

Reply:
@jonaslagoni Could you show an example? I am not sure that I get what you want to do.

Reply:
I do not like that the simplifier returns `0 ... n` models it should just return a simplified CommonModel for the schema and nothing more (this would reduce a lot of the complexity behind the simplifier as well). By removing the `ensureModelsAreSplit` function and `iteratedModels` property to a pre-processor we can do it after the schema has been simplified.

It of course means that we have to iterate the model twice, but I think it would be worth it in terms of maintainability.

Gonna add an issue clarifying this.

Reply:
https://github.com/asyncapi/generator-model-sdk/issues/195